src/main/java/Thea.java

```diff
@@ -0,0 +1,181 @@
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.Files;
+import java.io.BufferedWriter;
+import java.io.BufferedReader;
+import java.io.File;
+
+public class Thea {
+
+    public static void main(String[] args) throws Exception {
+        Scanner input = new Scanner(System.in);
+        ArrayList<Task> tasks =  retrieveTasks();
+        greet();
+        String userInput = input.nextLine();
+        while (true) {
+            String[] commandWords = userInput.split(" ", 2);
+            ArrayList<String> commandWordsArray = new ArrayList<>();
+            Collections.addAll(commandWordsArray, commandWords);
+            String command = commandWords[0];
+            if (command.equals("bye")) {
+                exit();
+                break;
+            } else if (command.equals("list")) {
+                printList(tasks);
+            } else if (command.equals("mark")) {
+                int index = Integer.parseInt(commandWords[1]) - 1;
+                if (index > tasks.size() - 1) {
+                    throw new IndexOutOfBoundsException("There is currently no task " + (index + 1));
+                } else {
+                    tasks.get(index).markAsDone();
+                    System.out.printf("Great job! ˊᗜˋ I've marked this task as done:\n  %s\n", tasks.get(index));
+                }
+            } else if (command.equals("unmark")) {
+                int index = Integer.parseInt(commandWords[1]) - 1;
+                if (index > tasks.size() - 1) {
+                    throw new IndexOutOfBoundsException("There is currently no task " + (index + 1));
+                } else {
+                    tasks.get(index).unmarkAsDone();
+                    System.out.printf("Okay, I've marked this task as not done yet:\n  %s\n", tasks.get(index));
+                }
+            } else if (command.equals("delete")){

```

Comment:
An alternative less tedious way might be using switch/case statements?

Reply:
I have updated the code! Thank you for your suggestion.